daft/dataframe/dataframe.py

```diff
@@ -889,6 +890,24 @@ def join(
         )
         return DataFrame(join_op)
 
+    @DataframePublicAPI
+    def drop_na(self, *cols):
+        """drops rows that contains NaNs or NULLs. If cols is None it will drop rows with any NULL value.
+        If column names are supplied, it will drop only those rows that contains NULLs in one of these columns.
+        Example:
+            >>> df = DataFrame.from_pydict({"a": [1, 2, 3, None]})
+            >>> df.drop_na()  # drops rows where any column contains Null/NaN values
+            >>> df = DataFrame.from_pydict({"a": [1, 2, 3, None]})
+            >>> df.drop_na("a")  # drops rows where columns x or y contain Null/NaN values
+
+        """
+        if len(cols) == 0:
+            cols = self.column_names
+        drop_na_op = logical_plan.Filter(
```

Comment:
Let's use `self.where(...)` instead to avoid duplicating logic for constructing the Filter node?

Reply:
Ah yes, you are totally right. I´ll change it accordingly.

Reply:
Changed to self.where